pysph/tools/ipy_viewer.py

```diff
@@ -727,6 +854,84 @@ def _save_all_plots_handler(self, change):
                 pa_widgets.vector_width.disabled = False
                 pa_widgets.vector_scale.disabled = False
                 pa_widgets.scalar_size.disabled = False
+                pa_widgets.is_visible.disabled = False
+
+    def _normalized_for_colormaps(self, data):
+        '''
+        Helper Function:
+        Takes in a numpy array and returns its maximum,
+        minimum and absolute maximum values, along with
+        the input array normalized by the absolute maximum.
+        '''
+
+        minm = np.min(data)
+        absmaxm = maxm = np.max(data)
+        if abs(maxm) < abs(minm):
+            absmaxm = abs(minm)
+        if absmaxm != 0:
+            normed_data = data/absmaxm
+            return [minm, maxm, absmaxm, normed_data]
+        else:
+            return [minm, maxm, absmaxm, data]
+
+    def _is_visible_handler(self, change):
+
+        array_name = change['owner'].owner
+        pa_widgets = self._widgets.particles[array_name]
+        temp_data = self.get_frame(self._widgets.frame.value)['arrays']
+        sct = self._scatters[array_name]
+
+        if change['new'] is False:
+            sct.set_offsets(None)
+        elif change['new'] is True:
+            sct.set_offsets(
+                np.vstack(
+                    (temp_data[array_name].x, temp_data[array_name].y)
+                ).T
+            )
+            c = getattr(
+                temp_data[array_name],
+                pa_widgets.scalar.value
+            )
+            colormap = getattr(
+                plt.cm,
+                pa_widgets.scalar_cmap.value
+            )
+            arr = self._normalized_for_colormaps(c)
```

Comment:
Cleanup.

Reply:
All of this has been changed in the latest commit.